turbinia/config/logger.py

```diff
@@ -62,7 +67,15 @@ def setup(need_file_handler=True, need_stream_handler=True):
               exception, config.CONFIG_MSG))
       sys.exit(1)
 
-    file_handler = logging.FileHandler(config.LOG_FILE)
+    # Check if user specified log name and update to default if not
+    if not user_specified_log:
+      log_name = uname().nodename
+      # Check if NODE_NAME available for GKE setups
+      if ENVNODENAME in environ:
+        log_name = log_name + '.{0!s}'.format(environ[ENVNODENAME])
+      config.LOG_DIR = '{0:s}/{1:s}.log'.format(config.LOG_DIR, log_name)
+
+    file_handler = logging.FileHandler(config.LOG_DIR)
```

Comment:
This is a little confusing to read without more context because the parameter should be a file path and it looks like we're passing in a directory.  See my comment below, but I think we might be able to change this to pass the new path into the method so we don't need to do this.  If we don't end up making that change we should at least put a comment or something here.

Reply:
Ty for review! So we call `logger.setup` in a few spots such as worker.py, server.py, and client.py (when we init each of them), which is why I'm creating the log path in `logger.setup`. Ex here: https://github.com/google/turbinia/blob/e358ac923bcbecdd6d4ffb76c5da0a2408af41c0/turbinia/worker.py#L57

Should I keep creating the log path in `logger.setup` or move that as well somehow? Or should I keep similar logic but just pass the user specified path in and set that as the log_name else create the new log_name.

Reply:
I think it's fine to create that in `logger.setup()`.  I was just thinking that it would be clearer to pass in the variable directly if possible.  Are you saying that we need to store the custom log path in `LOG_DIR` because this setup method could be called from other locations where `args.log_file` is not available?  If so, maybe we need to set `need_file_handler=False` in those other locations so that we don't override the file handler we create with the custom path?

Reply:
Ahh okay got it. Yes I see what you mean and agree. I went ahead and made some updates to parameter being passed in and setting file handler based on `log_file_path`. I also took a further look and stepped through the `args.log_file` parameter pass in and since we're setting this file path as a file handler, it actually isn't need/won't override since there is logic in `logger.setup` that checks for preexisting file handlers (which in the case user specified one, would have been set)